saleor/payment/utils.py

```diff
@@ -351,8 +351,14 @@ def _gateway_postprocess(transaction, payment):
     transaction_kind = transaction.kind
 
     if transaction_kind in [TransactionKind.CHARGE, TransactionKind.CAPTURE]:
-        payment.charge_status = ChargeStatus.CHARGED
         payment.captured_amount += transaction.amount
+
+        # Set payment charge status to fully charged
+        # only if there is no more amount needs to charge
+        payment.charge_status = ChargeStatus.PARTIALLY_CHARGED
```

Comment:
Why not putting it into the `else` block?

Reply:
I feels that It's more condense, which means ```PARTIALLY_CHARGED``` is by default, only set to ```FULLY_CHARGED``` under some circumstances.

Reply:
But we get to write into a variable first, then, we may override it again. That's bad as we write twice in memory, even though we will hit the CPU cache. It's wasting CPU cycles and I/O time for nothing :confused:

Reply:
It is just a simple string variable assignment instead of db operations or looping or large data structure assignment. I assume it is not heavy at all and I prefer a better readability.

Reply:
Yes, but the issues is that if `payment.get_charge_amount() <= 0` is `True`, we will have written twice in the variable. The issue with that is that putting the instruction in a simple `else` block would only be a simple jump CPU instruction, where only one write happen. Doing two write to maybe discard the first one is wasting important processing time.

Reply:
```
a = init_value
if condition:
    a = other_value
```

You could consider the first assignment is like variable initialization. It has better readability. A string assignment assignment probably only consumes one cpu cycle which is super fast. It is not like doing any I/O operations, db operations or looping. It is not writing embedded system and there is no need spending time in optimizing these cases.